spec/system/composer/category_templates_spec.rb

```diff
@@ -160,6 +160,77 @@
     sign_in user
   end
 
+  describe "discard draft modal" do
+    it "does not show the modal if there is no draft on a topic without a template" do
+      category_page.visit(category_no_template)
+      category_page.new_topic_button.click
+      composer.close
+      expect(composer).to be_not_opened
```

Comment:
`to be_not_opened` is... not easy to read. Can we do `.not_to be_opened` instead?

Reply:
No we shouldn't do this, it's not conceptually equivalent.

- For something to be considered not present it has TO NOT BE PRESET for the duration of the timeout

- For something TO BE NOT PRESENT, it has to not be there at the beginning of the check

Reply:
I see, thanks for the context @jjaffeux. 

Would writing an explicit condition, `not_be_opened?` work? If not, no worries, it's not a big concern. I am just a little offput by the readability of `be_not_opened`.

Reply:
yes writing custom matchers with page objects is the way to go to get nice assertions

Reply:
Yes I tried this, as well as just using the `closed` class that gets applied on the composer

```ruby
def closed?
  page.has_css?("#{COMPOSER_ID}.closed")
end
```

but I kept running into failures. I suspect the assertion `expect(composer).to be_closed` is just happening too fast, before things could change? I tried with `sleep 5` and but its flaky and only works sometimes...

Reply:
Yes, when I inspect the element it does have `closed` class.

I also tried logging the class list:

```ruby
$stdout.sync = true
result = page.evaluate_script("document.querySelector('#reply-control').classList;")
puts result
```

If I add the `sleep 5` then sometimes `closed` exists, but otherwise it has `open` class. Which is why I was suspecting it might just be happening too fast before things could change?

Reply:
Ah, I think I figured out why that isn't working @keegangeorge. It's because the element is invisible, and by default `has_css` is limited to visible elements. 

This works locally for me: 

```
      def closed?
        page.has_css?("#{COMPOSER_ID}.closed", visible: :all)
      end

```

See the documentation for more details: https://www.rubydoc.info/gems/capybara/Capybara%2FNode%2FFinders:all

Reply:
Aah I see, cool, thanks!